source/particles/particle_handler.cc

```diff
@@ -135,6 +143,14 @@ namespace Particles
 
     // Create the memory pool that will store all particle properties
     property_pool = std_cxx14::make_unique<PropertyPool>(n_properties);
+
+    // Update the triangulation cache to ensure that you can sort the particles
+    // correctly
+    update_triangulation_cache();
+
+    connection = triangulation->signals.any_change.connect(
```

Comment:
You already create this connection in the constructor, do you need it again here? At the moment this could result in two executions of `update_triangulation_cache` whenever `any_change` is triggered. If you need the additional `connect` here, you should call `connection.disconnect()` beforehand to remove existing connections (no harm if there is none before).